app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala

```diff
@@ -37,7 +37,7 @@ import scala.util.control.NonFatal
 @ImplementedBy(classOf[HelpToSaveConnectorImpl])
 trait HelpToSaveConnector {
 
-  def getEligibility()(implicit hc: HeaderCarrier): EitherT[Future, String, EligibilityCheckResult]
+  def getEligibility()(implicit hc: HeaderCarrier): EitherT[Future, String, EligibilityCheckResponse]
```

Comment:
The problem with returning `EligibilityCheckResponse` here is that the code in front is now too tightly coupled with  DES's interface with us. That interface is not very code friendly since it relies on an `Int` to convey eligibility, which is why later on in the code you have to do a match on `Ints` which is arbitrary when looked at in isolation. By converting the response we get from DES to something more sensibly typed it makes sure that we are less coupled to that interface and the code is more easily understood after this point.